tests/test_modeling_blenderbot.py

```diff
@@ -0,0 +1,426 @@
+          )
+      input_ids = torch.tensor(
+                  [
+                      [
+                          49,
+                          12,
+                          38,
+                          24,
+                          13,
+                          25,
+                          10,
+                          28,
+                          37,
+                          7,
+                          44,
+                          7,
+                          2,
+                          3
+                      ]
+                  ],
+                  dtype=torch.long,
+                  device=torch_device,
+              )
+      model = BlenderbotForConditionalGeneration(config)
+      model.eval()
+      model.to(torch_device)
+      # output from parlai model after copying the same blenderbot weight in parlai and setting a manual_seed
+      expected_logits = torch.tensor([[[-1.0000e+20,  0.0000e+00, -8.3858e-03,  5.3556e-02, -6.7345e-02,
+                                      -1.1861e-01, -4.7368e-02, -8.6005e-02, -6.6010e-02, -1.1263e-01,
+                                      -1.2138e-02, -5.0588e-02,  1.1818e-01,  3.8662e-03,  2.3491e-02,
+                                      -1.0256e-01,  1.9944e-02, -2.8050e-02,  1.2771e-01, -5.6630e-02,
+                                      -3.7779e-02,  6.9132e-02, -8.2159e-04, -6.3877e-02,  1.1591e-01,
+                                      9.1973e-02,  3.8424e-03,  5.4423e-02, -3.4574e-02,  3.1875e-02,
+                                      -3.2030e-02,  6.0317e-02,  6.8307e-02,  1.3964e-01, -1.2045e-01,
+                                      -1.1150e-01,  7.3168e-02, -4.0991e-02,  3.8692e-04,  5.9230e-02,
+                                      -2.0674e-02, -3.2628e-02, -9.5583e-02,  6.5901e-02,  5.8617e-02,
+                                      9.2186e-02, -4.5951e-02, -3.7279e-02, -1.5638e-02,  3.7328e-02]]])
+      decoder_inputs = torch.LongTensor([1]).expand(1,1).to(torch_device)
+      logits = model(input_ids, decoder_input_ids=decoder_inputs)['logits']
+    
+      assert torch.allclose(expected_logits, logits, atol=1e-4)
+      
+
+  @torch.no_grad()
+  @slow
+  def test_generation_from_long_input_same_as_parlai_90M(self):
+    src_text = [
+    "Social anxiety\nWow, I am never shy. Do you have anxiety?\nYes. I end up sweating and blushing and feel like i'm going to throw up.\nand why is that?"
+    ]
+    tgt_text = ["__start__ i ' m not sure . i just feel like i ' m going to throw up . __end__"]  
+
+    
+    model_inputs = self.tokenizer(src_text, return_tensors="pt").to(torch_device)
+    generated_utterances = self.model.generate(**model_inputs, min_length=15, length_penalty=0.65, max_length=128, early_stopping=True)
+   
+    self.assertListEqual(tgt_text, self.tokenizer.batch_decode(generated_utterances)) #test not passing yet generated_utterance = __start__ i don ' t know . i just feel like i ' m going to throw up .
+    
+  @torch.no_grad()
```

Comment:
I think we can delete `no_grad` everywhere. confirm that this doesn't change test speed.